File: docs_old/datatools/codesnippet/CSharp/save-data-in-a-transaction_1.cs
        private void customersBindingNavigatorSaveItem_Click(object sender, EventArgs e)
        {
            UpdateData();
        }

        private void UpdateData()
        {
            this.Validate();
            this.customersBindingSource.EndEdit();
            this.ordersBindingSource.EndEdit();

            using (System.Transactions.TransactionScope updateTransaction = 
                new System.Transactions.TransactionScope())
            {
                DeleteOrders();
                DeleteCustomers();
                AddNewCustomers();
                AddNewOrders();

                updateTransaction.Complete();
                northwindDataSet.AcceptChanges();
            }
        }

File: docs/data-tools/codesnippet/VisualBasic/save-data-in-a-transaction_1.vb
    Private Sub CustomersBindingNavigatorSaveItem_Click() Handles CustomersBindingNavigatorSaveItem.Click
        UpdateData()
    End Sub

    Private Sub UpdateData()
        Me.Validate()
        Me.CustomersBindingSource.EndEdit()
        Me.OrdersBindingSource.EndEdit()

        Using updateTransaction As New Transactions.TransactionScope

            DeleteOrders()
            DeleteCustomers()
            AddNewCustomers()
            AddNewOrders()

            updateTransaction.Complete()
            NorthwindDataSet.AcceptChanges()
        End Using
    End Sub